lib/plugins/aws/package/compile/events/lib/ensureApiGatewayCloudWatchRole.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const { memoize } = require('lodash');
+const { addCustomResourceToService } = require('../../../../customResources');
+
+module.exports = memoize(provider => {
+  const cfTemplate = provider.serverless.service.provider.compiledCloudFormationTemplate;
+  const customResourceLogicalId = provider.naming.getCustomResourceApiGatewayAccountCloudWatchRoleResourceLogicalId();
+  const customResourceFunctionLogicalId = provider.naming.getCustomResourceApiGatewayAccountCloudWatchRoleHandlerFunctionLogicalId();
+
+  cfTemplate.Resources[customResourceLogicalId] = {
+    Type: 'Custom::ApiGatewayAccountRole',
+    Version: 1.0,
+    Properties: {
+      ServiceToken: {
+        'Fn::GetAtt': [customResourceFunctionLogicalId, 'Arn'],
+      },
+    },
+  };
+
+  return addCustomResourceToService(provider, 'apiGatewayCloudWatchRole', [
+    {
+      Effect: 'Allow',
+      Resource: {
+        'Fn::Join': [':', ['arn:aws:iam:', { Ref: 'AWS::AccountId' }, 'role/*']],
+      },
+      Action: [
+        'iam:AttachRolePolicy',
+        'iam:CreateRole',
```

Comment:
I know this PR has been merged about 10 days go.

I run AWS account on lockdown environment, where `iam:CreateRole` is not allowed.

I am just wondering why do you need `iam:CreateRole` permission?

Reply:
We need to assign a IAM role in API Gateway settings, so CloudWatch logs can be successfully written.
 So if there's no role assigned there, we create one first that has correct permissions and assign it.

Does it crashes for you simply because this permission is listed? Note that custom resource will not attempt to create a role if you already have a CloudWatch role setup on API Gateway (for stages you're using)

Reply:
It fails completely.
See: https://gist.github.com/azizur/6669ce32634f5ab316f0d91d03ec6ca4

Reply:
Then indeed it's an issue, we should provide some option to opt-out. Can you open a dedicated issue for it?